tests/gpflow/posteriors/test_posteriors.py

```diff
@@ -0,0 +1,378 @@
+    )
+    assert isinstance(posterior, FallbackIndependentLatentPosterior)
+
+    _assert_fused_predict_f_equals_precomputed_predict_f(posterior, full_cov, full_output_cov)
+
+
+def test_fallback_independent_multi_output_shi(
+    set_q_sqrt, full_cov, full_output_cov, whiten, output_dims,
+):
+    """
+    Fallback posterior with shared independent inducing variables.
+
+    The FallbackIndependentLatentPosterior is a subclass of the FullyCorrelatedPosterior which
+    requires a single latent GP function.
+    """
+    kernel = gpflow.kernels.LinearCoregionalization(
+        [gpflow.kernels.SquaredExponential()], W=tf.ones((output_dims, 1))
+    )
+    inducing_variable = gpflow.inducing_variables.FallbackSharedIndependentInducingVariables(
+        inducingpoint_wrapper(np.random.randn(NUM_INDUCING_POINTS, INPUT_DIMS))
+    )
+
+    q_mu = np.random.randn(NUM_INDUCING_POINTS, 1)
+
+    q_sqrt = None
+    if set_q_sqrt:
+        q_sqrt = tf.eye(NUM_INDUCING_POINTS, batch_shape=[1], dtype=tf.float64)
+
+    posterior = create_posterior(
+        kernel=kernel, inducing_variable=inducing_variable, q_mu=q_mu, q_sqrt=q_sqrt, whiten=whiten,
+    )
+    assert isinstance(posterior, FallbackIndependentLatentPosterior)
+
+    _assert_fused_predict_f_equals_precomputed_predict_f(posterior, full_cov, full_output_cov)
+
+
+TESTED_POSTERIORS.add(LinearCoregionalizationPosterior)
+
+
+def test_linear_coregionalization_sei(
+    set_q_sqrt, full_cov, full_output_cov, whiten, num_latent_gps, output_dims,
+):
+    """
+    Linear coregionalization posterior with separate independent inducing variables.
+    """
+    kernel = gpflow.kernels.LinearCoregionalization(
+        [gpflow.kernels.SquaredExponential() for _ in range(num_latent_gps)],
+        W=tf.ones((output_dims, num_latent_gps)),
+    )
+    inducing_variable = gpflow.inducing_variables.SeparateIndependentInducingVariables(
+        [
+            inducingpoint_wrapper(np.random.randn(NUM_INDUCING_POINTS, INPUT_DIMS))
+            for _ in range(num_latent_gps)
+        ]
+    )
+
+    q_mu = np.random.randn(NUM_INDUCING_POINTS, num_latent_gps)
+
+    q_sqrt = None
+    if set_q_sqrt:
+        q_sqrt = tf.eye(NUM_INDUCING_POINTS, batch_shape=[num_latent_gps], dtype=tf.float64)
+
+    posterior = create_posterior(
+        kernel=kernel, inducing_variable=inducing_variable, q_mu=q_mu, q_sqrt=q_sqrt, whiten=whiten,
+    )
+    assert isinstance(posterior, LinearCoregionalizationPosterior)
+
+    _assert_fused_predict_f_equals_precomputed_predict_f(posterior, full_cov, full_output_cov)
+
+
+def test_linear_coregionalization_shi(
+    set_q_sqrt, full_cov, full_output_cov, whiten, num_latent_gps, output_dims,
+):
+    """
+    Linear coregionalization with shared independent inducing variables.
+    """
+    kernel = gpflow.kernels.LinearCoregionalization(
+        [gpflow.kernels.SquaredExponential() for _ in range(num_latent_gps)],
+        W=tf.ones((output_dims, num_latent_gps)),
```

Comment:
would be good to use something more random for `W`

Reply:
I will replace `tf.ones` with `tf.random.normal`.